### PR TITLE
Array is now dynamic

### PR DESCRIPTION
**Pull request by brianwchou**
## As the title says the stack is now dynamic

### CODE REVIEW

File: contracts/Stack.sol [hunk 1]
```diff
@@ -1,9 +1,8 @@
 pragma solidity >=0.4.25 <0.6.0;
 
 contract Stack {
-    uint[10] private array;
+    uint[] private array;
     uint private counter = 0;
```
Comment (ejwessel) [inline contracts/Stack.sol]: remove counter. You can access array length directly with `.length`

File: contracts/Stack.sol [hunk 2]
```diff
@@ -1,9 +1,8 @@
 pragma solidity >=0.4.25 <0.6.0;
 
 contract Stack {
-    uint[10] private array;
+    uint[] private array;
```
Comment (ejwessel) [inline contracts/Stack.sol]: while we're here, rename `array` to something more descriptive.

**Review by ejwessel — CHANGES_REQUESTED** [review]
Changes

**Comment by ejwessel** [comment]
@brianwchou lets also add 
https://www.npmjs.com/package/ganache-time-traveler

File: contracts/Stack.sol (hunk 1 above)
Comment (brianwchou) [reply contracts/Stack.sol]: I tested push and it permanently increments length. So if I use push and then pop on an empty stack the array will be size 1. I think you need counter.

[explanation](https://ethereum.stackexchange.com/questions/58495/understanding-more-about-delete-keyword-in-solidity)

Comment (ejwessel) [reply contracts/Stack.sol]: try this
https://ethereum.stackexchange.com/questions/13343/setting-the-length-property-of-array

Comment (brianwchou) [reply contracts/Stack.sol]: That is wild.

Okay I've made the edits

File: contracts/Stack.sol [hunk 3]
```diff
@@ -1,31 +1,26 @@
...
+        return data[data.length - 1];
     }
 
     function pop() public {
-        require(counter > 0, "stack needs to have values");
+        require(data.length > 0, "stack needs to have values");
 
-        delete array[counter - 1];
-        counter -= 1;
+        delete data[data.length - 1];
```
Comment (ejwessel) [inline contracts/Stack.sol]: I believe you can shorten the length without needing to delete the data. This effectively makes the value unreachable and also saves gas.

**Review by ejwessel — CHANGES_REQUESTED** [review]
Last change requested.

File: contracts/Stack.sol (hunk 1 above)
Comment (ejwessel) [reply contracts/Stack.sol]: Agreed when I had the same reaction when I first learned it too.

**Review by ejwessel — APPROVED** [review]
LGTM

**Pull request merged**